tapgarden/caretaker.go

```diff
@@ -1188,3 +1189,104 @@ func GenHeaderVerifier(ctx context.Context,
 		return err
 	}
 }
+
+// GenGroupVeifier generates a group key verification callback function given
+// a DB handle.
+func GenGroupVerifier(ctx context.Context,
+	mintingStore MintingStore) func(*btcec.PublicKey) error {
+
+	return func(groupKey *btcec.PublicKey) error {
+		if groupKey == nil {
+			return fmt.Errorf("cannot verify empty group key")
+		}
+
+		// This query will err if no stored group has a matching
+		// tweaked group key.
+		_, err := mintingStore.FetchGroupByGroupKey(
+			ctx, groupKey,
+		)
+		if err != nil {
+			assetGroupKey := groupKey.SerializeCompressed()
+			return fmt.Errorf("group %x not imported: %w",
+				assetGroupKey, err)
+		}
+
+		return nil
+	}
+}
+
+// GenGroupAnchorVerifier generates a caching group anchor verification callback
+// function given a DB handle.
+func GenGroupAnchorVerifier(ctx context.Context,
+	mintingStore MintingStore) func(*asset.Genesis,
+	*asset.GroupKey) error {
+
+	// Cache anchors for groups that were previously fetched.
+	groupAnchors := make(map[asset.SerializedKey]*asset.Genesis)
+
+	return func(gen *asset.Genesis, groupKey *asset.GroupKey) error {
+		assetGroupKey := asset.ToSerialized(&groupKey.GroupPubKey)
+		groupAnchor, ok := groupAnchors[assetGroupKey]
+
+		if !ok {
+			storedGroup, err := mintingStore.FetchGroupByGroupKey(
+				ctx, &groupKey.GroupPubKey,
+			)
+			if err != nil {
+				return err
+			}
+
+			groupAnchors[assetGroupKey] = storedGroup.Genesis
+			groupAnchor = storedGroup.Genesis
+		}
+
+		if gen.ID() != groupAnchor.ID() {
```

Comment:
Ah ok, so this verifies that the given asset is actually the root/anchor asset for the group key. In other words, we can use this to verify if an asset is the first asset to be issued with a group key.